sopel/modules/announce.py

```diff
@@ -12,12 +12,26 @@
 from sopel import plugin
 
 
+def _chunks(items, size):
+    """Break a list of items into groups.
+
+    :param items: the collection of items to chunk
+    :type items: :term:`iterable`
+    :param int size: the size of each chunk
+    """
+    # from https://stackoverflow.com/a/312464/5991 with modified names for readability
+    for delim in range(0, len(items), size):
+        yield items[delim:delim + size]
+
+
 @plugin.command('announce')
 @plugin.example('.announce Some important message here')
 @plugin.require_admin('Sorry, I can\'t let you do that', reply=True)
 @plugin.output_prefix('[ANNOUNCEMENT] ')
 def announce(bot, trigger):
     """Send an announcement to all channels the bot is in"""
-    for channel in bot.channels:
-        bot.say(trigger.group(2), channel)
+    size = bot.isupport.get('TARGMAX', {}).get('PRIVMSG', 1)
+    channels = _chunks(bot.channels.keys(), size)
+    for cgroup in channels:
+        bot.say(trigger.group(2), ','.join(cgroup))
```

Comment:
If you could check that there is a message here, that would do it for #1001 as well!

Reply:
Should be satisfactorily implemented in ce10d0a